src/Gaffer/Spreadsheet.cpp

```diff
@@ -496,22 +614,63 @@ void Spreadsheet::hash( const ValuePlug *output, const Context *context, IECore:
 
 void Spreadsheet::compute( ValuePlug *output, const Context *context ) const
 {
-	if( output == rowIndexPlug() )
+	if( output == rowsMapPlug() )
+	{
+		StringVectorDataPtr activeRowNamesData = new StringVectorData;
+		vector<string> &activeRowNames = activeRowNamesData->writable();
+		RowsMapPtr result = new RowsMap;
+		result->activeRowNames = activeRowNamesData;
+
+		for( size_t i = 1, e = rowsPlug()->children().size(); i < e; ++i )
+		{
+			const auto *row = rowsPlug()->getChild<RowPlug>( i );
+			if( !row->enabledPlug()->getValue() )
+			{
+				continue;
+			}
+
+			const std::string name = row->namePlug()->getValue();
+			activeRowNames.push_back( name );
+
+			if(
+				StringAlgo::hasWildcards( name ) ||
+				name.find( ' ' ) != string::npos
+			)
+			{
+				result->wildcardRows.push_back( { name, i } );
+			}
+			else
+			{
+				result->plainRows.insert( { name, i } );
+			}
+		}
+
+		static_cast<ObjectPlug *>( output )->setValue( result );
+		return;
+	}
+	else if( output == rowIndexPlug() )
 	{
-		int result = 0;
+		size_t result = 0;
 		if( enabledPlug()->getValue() )
 		{
-			const std::string selector = selectorPlug()->getValue();
-			for( int i = 1, e = rowsPlug()->children().size(); i < e; ++i )
+			RowsMapScope rowsMapScope( context, selectorPlug()->getValue() );
+			ConstRowsMapPtr rowsMap = boost::static_pointer_cast<const RowsMap>( rowsMapPlug()->getValue() );
+
+			RowsMap::Map::const_iterator it = rowsMap->plainRows.find( rowsMapScope.selector() );
+			if( it != rowsMap->plainRows.end() )
 			{
-				const auto *row = rowsPlug()->getChild<RowPlug>( i );
-				if( !row->enabledPlug()->getValue() )
+				result = it->second;
+			}
+			for( auto &row : rowsMap->wildcardRows )
+			{
+				if( result && row.index > result )
```

Comment:
The only improvement I could think of might be to add a quick note to say that `result` will never be `0` for any actual row as index `0` is the `default` row....